diff --git a/server.hpp b/server.hpp
--- a/server.hpp
+++ b/server.hpp
@@ -40,7 +40,10 @@ typedef struct s_request
     std::string body;
     size_t content_length;
     size_t body_received;
-    s_request(): server(NULL), content_length(0),body_received(0){}
+    bool is_chunked;
+    std::string body_file;
+    int body_fd;
+    s_request(): server(NULL), content_length(0),body_received(0), is_chunked(false), body_fd(-1){}
 
 }t_request;
 
@@ -52,7 +55,8 @@ typedef struct s_respond
     std::string body;
     std::string outbuf;
     size_t sent;
-    s_respond() : status_code(0), sent(0){}
+    size_t bytes_sent;
+    s_respond() : status_code(0), sent(0), bytes_sent(0){}
 
 }t_response;
